lib/iris/_merge.py

```diff
@@ -441,6 +450,11 @@ def match(self, other, error_on_mismatch):
             msgs.append(msg.format(self.data_type, other.data_type))
         if self.cell_measures_and_dims != other.cell_measures_and_dims:
             msgs.append("cube.cell_measures differ")
+        if (
+            self.ancillary_variables_and_dims
+            != other.ancillary_variables_and_dims
```

Comment:
I think it could be worth emphasizing here that although it looks like these two av-and-dims lists might differ only in their occurrence order, in fact this is ok as they are always stored in a cube in a specific sorted order, through [this code](https://github.com/SciTools/iris/blob/479b2cf80fc75c7b327c01a0f6a2b139ea6e9896/lib/iris/cube.py#L1130)

It caused me some thought, anyway !